api/v1/controllers/EventController.js

```diff
@@ -0,0 +1,154 @@
+}
+
+function createEvent(req, res, next) {
+        return services.EventService.createEvent(req.body.name,
+                                                 req.body.shortName,
+                                                 req.body.description,
+                                                 req.body.qrCode,
+                                                 req.body.startTime,
+                                                 req.body.endTime,
+                                                 req.body.tag || "SCHEDULE",
+                                                 req.body.locations || [])
+                .then(function(result) {
+                        res.body = result.toJSON();
+                        next();
+                        return null;
+                });
+}
+
+function createLocation(req, res, next) {
+        return services.EventService.createLocation(null,
+                                                    req.body.name,
+                                                    req.body.shortName,
+                                                    req.body.latitude,
+                                                    req.body.longitude)
+                .then(function (result) {
+                        res.body = result.length == 0 ? [] : result.toJSON();
+                        next();
+                        return null;
+                });
+}
+
+function getLocations(req, res, next) {
+        return services.EventService.getLocations()
+                .then(function (result) {
+                        res.body = result.length == 0 ? [] : result.toJSON();
+                        next();
+                        return null;
+                });
+}
+
+function addLocationToEvents(req, res, next) {
+        return services.EventService.addLocationToEvents(req.body.eventIds,
+                                                         req.body.locationId)
+                .then(function (event) {
+                        next();
+                        return null;
+                })
+                .catch(function (err) {
+                        next(err);
+                        return null;
+                });
+}
+
+function addEventToLocations(req, res, next) {
+        return services.EventService.addEventToLocations(req.body.eventId,
+                                                         req.body.locationIds)
+                .then(function (result) {
+                        return services.EventService.findEventById(req.body.eventId);
+                })
+                .then(function (event) {
+                        res.body = event.toJSON();
+                        next();
+                        return null;
+                })
+                .catch(function (err) {
+                        next(err);
+                        return null;
+                });
+}
+
+function deleteReference(req, res, next) {
+        if (!req.body.eventIds && !req.body.locationIds)
```

Comment:
sadly no, it doesn't seem like the validation has granularity to declare that "both eventIds or locationIds are optional, but at least one of them must exist"